internal/clientcache/internal/db/db.go

```diff
@@ -68,3 +97,56 @@ func createTables(ctx context.Context, conn *db.DB) error {
 	}
 	return nil
 }
+
+// validSchema checks of the schema is valid based on its version. Options
+// supported: withTestValidSchemaVersion (for testing purposes)
+func validSchema(ctx context.Context, conn *db.DB, opt ...Option) (bool, error) {
+	const op = "validateSchema"
+	switch {
+	case conn == nil:
+		return false, errors.New(ctx, errors.InvalidParameter, op, "conn is missing")
+	}
+	opts, err := getOpts(opt...)
+	if err != nil {
+		return false, errors.Wrap(ctx, err, op)
+	}
+	if opts.withSchemaVersion == "" {
+		opts.withSchemaVersion = schemaCurrentVersion
+	}
+
+	rw := db.New(conn)
+	s := schema{}
+	err = rw.LookupWhere(ctx, &s, "1=1", nil)
+	switch {
+	case err != nil && strings.Contains(err.Error(), "no such table: schema_version"):
+		return false, nil
+	case err != nil:
+		// not sure if we should return the error or just return false so the
+		// schema is recreated... for now return the error.
+		return false, fmt.Errorf("%s: unable to get version: %w", op, err)
+	case s.Version != opts.withSchemaVersion:
+		return false, nil
+	default:
+		return true, nil
+	}
+}
+
+// schema represents the current schema in the database
+type schema struct {
+	// Version of the schema
+	Version string
+	// UpdateTime is the last update of the version
+	UpdateTime time.Time
+	// CreateTime is the create time of the initial version
+	CreateTime time.Time
+}
+
+const (
+	schemaTableName      = "schema_version"
+	schemaCurrentVersion = "v0.0.1"
```

Comment:
Should we not just use incrementing integers for this? I can't imagine semantic versioning making sense for database schemas, though in practice I guess it just needs to be a different string from any previous version.

Reply:
this is prior art from go-kms-wrapping/extras and I think it's sufficient.